### PR TITLE
CEDS 1187

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/services/model/AutoCompleteItem.scala [hunk 1]
```diff
@@ -29,4 +29,8 @@ object AutoCompleteItem {
 
   def fromDocumentType(documents: List[DocumentType]): List[AutoCompleteItem] =
     documents map (d => AutoCompleteItem(s"${d.description} - ${d.code}", d.code))
+
+  def fromNationalAdditionalCode(codes: List[NationalAdditionalCode]): List[AutoCompleteItem] = {
+    codes map (c => AutoCompleteItem(c.value, c.value))
+  }
```
Comment (PatrykRudnicki) [inline app/services/model/AutoCompleteItem.scala]: Only style comment, but unnecessary parenthesis. It'll be good to keep convention like above for similar methods.

File: app/views/components/fields/field_multiple_values.scala.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @if(elements.nonEmpty) {
+        <div class="field-group">
+            <table >
+            @elements.zipWithIndex.map { case (elem, index) =>
+                <tr>
+                    <th>@elem</th>
+                    <th>
+                        <button
+                            class="button--secondary"
+                            name="@Remove.toString"
```
Comment (PatrykRudnicki) [inline app/views/components/fields/field_multiple_values.scala.html]: I'm not sure about it but it's better to ask than skip. I think we should wrap `Remove.toString` in parenthesis. Like "@{Remove.toString}"

**Comment by ed0906** [comment]
For anyone doing autocomplete stories, I've changed the name of the component to `field_autocomplete` and moved it into the `components.fields` package

File: app/utils/validators/forms/supplementary/ItemTypeValidator.scala [hunk 3]
```diff
@@ -67,14 +68,13 @@ object ItemTypeValidator extends Validator[ItemType] {
...
+        "declaration.itemType.nationalAdditionalCode.error.invalid",
+        isContainedIn(NationalAdditionalCode.all.map(_.value))
       )
-      .verifying("declaration.itemType.nationalAdditionalCode.error.specialCharacters", isAlphanumeric)
   ).verifying(
-      "declaration.itemType.nationalAdditionalCode.error.maxAmount",
-      codes => codes.size <= nationalAdditionalCodesMaxAmount
-    )
+    "declaration.itemType.nationalAdditionalCode.error.maxAmount",
+    codes => codes.size <= nationalAdditionalCodesMaxAmount
```
Comment (ed0906) [inline app/utils/validators/forms/supplementary/ItemTypeValidator.scala]: Not sure this is technically needed as there isnt even 99 codes to use but I've left it in incase the csv grows in the future

File: app/views/components/fields/field_multiple_values.scala.html (hunk 2 above)
Comment (ed0906) [reply app/views/components/fields/field_multiple_values.scala.html]: Fixed

File: app/services/model/AutoCompleteItem.scala (hunk 1 above)
Comment (ed0906) [reply app/services/model/AutoCompleteItem.scala]: Fixed

**Pull request merged**